acceptance-tests/lua_test.go

```diff
@@ -0,0 +1,69 @@
+package acceptance_tests
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("Lua scripting", func() {
+	It("Deploys haproxy with lua script", func() {
+		replyLuaTargetPath := "/var/vcap/packages/haproxy/lua_test.lua"
+		opsfileLua := fmt.Sprintf(`---
+		# Enable Lua scripting
+		- type: replace
+			path:  /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/lua_scripts?
+			value: %s
+		- type: replace
+		  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/frontend_config?
+		  value: |-
+			http-request use-service lua.lua_test if { path /lua_test }
+			`, replyLuaTargetPath)
+
+		replyLuaContent := `
+local function lua_test(applet)
+    -- If client is POSTing request, receive body
+    -- local request = applet:receive()
+
+    local response = string.format([[
+        <html>
+            <body>Running Lua %s</body>
+        </html>
+    ]], _VERSION)
+
+    applet:set_status(200)
+    applet:add_header("content-length", string.len(response))
+    applet:add_header("content-type", "text/html")
+	applet:add_header("lua-version", _VERSION)
+    applet:start_response()
+    applet:send(response)
+end
+
+core.register_service("lua_test", "http", lua_test)
+		`
+
+		haproxyBackendPort := 12000
+		haproxyInfo, _ := deployHAProxy(baseManifestVars{
+			haproxyBackendPort:    haproxyBackendPort,
+			haproxyBackendServers: []string{"127.0.0.1"},
+			deploymentName:        deploymentNameForTestNode(),
+		}, []string{opsfileLua}, map[string]interface{}{}, true)
+
+		// upload Lua script file
+		uploadFile(haproxyInfo, strings.NewReader(replyLuaContent), replyLuaTargetPath)
+
+		closeLocalServer, localPort := startDefaultTestServer()
```

Comment:
you might not even need a test server because you don't need a backed. the health check will be down but you can ignore that when calling lua.

Reply:
It won't harm either, correct? I'm not sure how well the setup handles not having the backend and it's difficult to test locally right now.

Reply:
it makes the test longer. not much but it counts.
just now this:
the way you are starting haproxy now it will report unhealthy anyways until the healthchecks to the backend had a chance to run a couple times. so without the `eventually` closure like in other tests, by the time your code (calling lua) runs, haproxy will still be unhealthy. so adding the server without waiting for health to turn green is actually pointless.

Reply:
ok, fair. I was worried it would not start HAProxy if it can't reach the backend. Not starting is only the case when it can't _resolve_ the address though. I will take it out.

Reply:
For completeness: the backend is back in as there is no way in the current test harness to disable it (easily). We will look into having an HAProxy without backend in the future.